deploy/crds/operator.tigera.io_authentications_crd.yaml

```diff
@@ -0,0 +1,91 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: authentications.operator.tigera.io
+spec:
+  group: operator.tigera.io
+  names:
+    kind: Authentication
+    listKind: AuthenticationList
+    plural: authentications
+    singular: authentication
+  scope: Cluster
+  versions:
+  - name: v1
+    schema:
+      openAPIV3Schema:
+        description: Authentication is the Schema for the authentications API
+        properties:
+          apiVersion:
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#resources'
+            type: string
+          kind:
+            description: 'Kind is a string value representing the REST resource this
+              object represents. Servers may infer this from the endpoint the client
+              submits requests to. Cannot be updated. In CamelCase. More info: https://git.k8s.io/community/contributors/devel/api-conventions.md#types-kinds'
+            type: string
+          metadata:
+            type: object
+          spec:
+            description: AuthenticationSpec defines the desired state of Authentication
+            properties:
+              managerDomain:
+                description: ManagerDomain is the domain name of the Manager
+                type: string
+              method:
+                description: 'Method configures the method of authentication used
+                  by Kibana. Default: Basic'
+                enum:
+                - Basic
+                - OIDC
+                type: string
+              oidc:
+                description: OIDC contains the configuration needed to setup OIDC
+                  authentication. If the method is OIDC then this is required, if
+                  the method is not OIDC then this must not be specified.
+                properties:
+                  groupsClaim:
+                    description: GroupsClaim specifies which claim to use from the
+                      OIDC provider as the group
+                    type: string
+                  groupsPrefix:
+                    description: UsernamePrefix is used to identify a ClusterRoleBinding
```

Comment:
s/Username/Groups/.  I think the wording is a bit off too: not sure what "that has a prefix with the authenticated user" means.

It's an awkward one to document since it sounds like ES doesn't really support it in the way it should be - bringing the CRBs into it perhaps adds confusion.  We'll also eventually want the field to be used in the conventional way (Kubernetes like) when we do auth.

"If specified, GroupsPrefix is prepended to each group obtained from the claims specified by GroupsClaim.  Note that Kibana does not support a groups prefix, so this prefix is removed from Kubernetes Groups when translating log access ClusterRoleBindings into Elastic." maybe?

Reply:
That's a perfect description, thanks!